libmamba/src/api/pip_utils.cpp

```diff
@@ -0,0 +1,166 @@
+#include <fmt/color.h>
+#include <fmt/format.h>
+#include <fmt/ostream.h>
+#include <fmt/ranges.h>
+#include <reproc++/run.hpp>
+#include <reproc/reproc.h>
+
+#include "mamba/api/install.hpp"
+#include "mamba/core/activation.hpp"
+#include "mamba/core/context.hpp"
+#include "mamba/core/util.hpp"
+#include "mamba/fs/filesystem.hpp"
+#include "mamba/util/environment.hpp"
+
+#include "tl/expected.hpp"
+
+#include "pip_utils.hpp"
+
+namespace mamba
+{
+    namespace
+    {
+        tl::expected<command_args, std::runtime_error> get_pip_install_command(
+            const std::string& name,
+            const std::string& target_prefix,
+            const fs::u8path& spec_file,
+            bool update
+        )
+        {
+            const auto python = util::which_in("python", get_path_dirs(target_prefix)).string();
+
+            const std::unordered_map<std::string, command_args> pip_install_command{
+                { "pip",
+                  { python, "-m", "pip", "install", "-r", spec_file, "--no-input", "--quiet" } },
+                { "pip --no-deps",
+                  { python, "-m", "pip", "install", "--no-deps", "-r", spec_file, "--no-input", "--quiet" } }
+            };
+
+            const std::unordered_map<std::string, command_args> pip_update_command{
+                { "pip",
+                  { python, "-m", "pip", "install", "-U", "-r", spec_file, "--no-input", "--quiet" } },
+                { "pip --no-deps",
+                  { python, "-m", "pip", "install", "-U", "--no-deps", "-r", spec_file, "--no-input", "--quiet" } }
+            };
+
+            auto found_it = update ? pip_update_command.find(name) : pip_install_command.find(name);
```

Comment:
I wonder if something like:
```cpp
const std::unordered_map<std::string, command_arg> pip_command{
    { "pip",
      { python, "-m", "pip", "install", (update ? ("-U", "-r") : "-r"), spec_file, "--no-input", "--quiet" } }, 
      // ...
};

auto found_it = pip_command.find(name);
```

would work. I t avoids some code duplication and the creation of two maps.